src/main/java/liquibase/ext/databricks/snapshot/jvm/SchemaSnapshotGeneratorDatabricks.java

```diff
@@ -16,113 +17,35 @@
 import liquibase.structure.core.Data;
 import liquibase.structure.core.Schema;
 import liquibase.util.JdbcUtils;
+import liquibase.util.JdbcUtil;
 
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
 
 public class SchemaSnapshotGeneratorDatabricks extends SchemaSnapshotGenerator {
-
     @Override
     public int getPriority(Class<? extends DatabaseObject> objectType, Database database) {
-        int priority = super.getPriority(objectType, database);
-        if (priority > PRIORITY_NONE && database instanceof DatabricksDatabase) {
-            priority += DatabricksDatabase.DATABRICKS_PRIORITY_DATABASE;
+        if (database instanceof DatabricksDatabase) {
+            return super.getPriority(objectType, database) + PRIORITY_DATABASE;
+        } else {
+            return PRIORITY_NONE;
         }
-        return priority;
     }
 
     @Override
     protected String[] getDatabaseSchemaNames(Database database) throws SQLException, DatabaseException {
         List<String> returnList = new ArrayList<>();
 
-        ResultSet schemas = null;
-        try {
-            schemas = ((JdbcConnection) database.getConnection()).getMetaData()
-                    .getSchemas(database.getDefaultCatalogName(), null);
-
+        try (ResultSet schemas = ((JdbcConnection) database.getConnection()).getMetaData().getSchemas(database
+                .getDefaultCatalogName(), null)) {
             while (schemas.next()) {
-                returnList.add(JdbcUtils.getValueForColumn(schemas, "TABLE_SCHEM", database));
-            }
-        } finally {
-            if (schemas != null) {
-                schemas.close();
-            }
-        }
-
-        return returnList.toArray(new String[returnList.size()]);
-    }
-
-    @Override
-    protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException, InvalidExampleException {
```

Comment:
This method should be really gone?

Reply:
Its an override, so its not gone, but this was the reason things were breaking, it wasnt able to find the schema and tables correctly when it did a snapshot - and returned an empty snapshot of table/column objects. Removing this custom override fixed it.